### PR TITLE
Disable shortcut l10n during shortcut processing

### PR DESCRIPTION
**Pull request by sangwoo108**
On Mac, we can't parse shortcut from keyEquivalent properly when the keyEquivalent contains localized string.

Resolves https://github.com/brave/brave-browser/issues/31770

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Reproducible on OS Ventura
* Select Czech-qwerty as  keyboard layout 
* Launch brave browser
* Check if `cmd + left arrow` works as `history back`
* Check if `cmd + =` works as `zoom plus`

### CODE REVIEW

**Comment by sangwoo108** [comment]
@fallaciousreasoning You saved my day! It looks working

**Comment by fallaciousreasoning** [comment]
Wow awesome! I'm so glad it worked - I was worried we were going to fall down a deep, deep rabbit hole :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
LGTM - thanks heaps for working all this out

File: browser/ui/views/commands/default_accelerators_mac.mm [hunk 1]
```diff
@@ -50,6 +50,13 @@ bool CanConvertToAcceleratorMapping(NSMenuItem* item) {
 }
 
 AcceleratorMapping ToAcceleratorMapping(NSMenuItem* item) {
+  if (@available(macos 12.0, *)) {
+    // We can't parse keyEquivalent into keycode properly when it's unicode
+    // character. So before starting parsing, disable l10n for a while.
+    // https://github.com/brave/brave-browser/issues/31770
+    item.allowsAutomaticKeyEquivalentLocalization = NO;
```
Comment (simonhong) [inline browser/ui/views/commands/default_accelerators_mac.mm:57]: According to the docs(https://developer.apple.com/documentation/appkit/nsmenuitem/3787554-allowsautomatickeyequivalentloca?language=objc), setting `NO` here disables `allowsAutomaticKeyEquivalentMirroring` also.
Don't we need to `allowsAutomaticKeyEquivalentMirroring` value reset also when we set `YES` to `allowsAutomaticKeyEquivalentLocalization`

Comment (sangwoo108) [reply browser/ui/views/commands/default_accelerators_mac.mm]: Good point! Changed to reset it together.

**Review by simonhong — APPROVED** [review]
++ 👍🏼

**Comment by fallaciousreasoning** [comment]
@sangwoo108 is it okay if I create uplifts for this to 1.56 & 1.57?

**Comment by stephendonner** [comment]
While on a working vacation, I didn't bring my **_Ventura_**-based macBook Pro (just my **_Big Sur_** `x64` and `arm64` machines).  

Sorry, but *I* won't be able to test this for uplift until Friday, if we can wait until then?

**Comment by kjozwiak** [comment]
> While on a working vacation, I didn't bring my **_Ventura_**-based macBook Pro (just my **_Big Sur_** `x64` and `arm64` machines).
> 
> Sorry, but _I_ won't be able to test this for uplift until Friday, if we can wait until then?

@stephendonner we'll need this for tomorrow's `C115` release. I'll run through the STR/Cases mentioned https://github.com/brave/brave-core/pull/19411#issue-1819354999 on my `Ventura` MBP.

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

Brave | 1.58.55 Chromium: 116.0.5845.51 (Official Build) nightly (x86_64)
-- | --
Revision | 34193c8918bf9f572cebc0ac8bf4a5ba1a102483
OS | macOS Version 14.0 (Build 23A5301g)

### Steps:
1. clicked `Apple` menu
2. clicked `System Settings...`
3. clicked `Accessibility`
4. clicked `Keyboard`
5. under `Text Input`, clicked `Edit...`
6. clicked `+`
7. selected `Czech - QWERTY`
8.

**Pull request merged**